src/server.rs

```diff
@@ -397,8 +397,14 @@ impl<O: Output> LsService<O> {
         self.output.success(id, ResponseData::Init(result));
 
         let root_path = params.root_path.map(PathBuf::from).expect("No root path");
+        let init_options: Option<InitializationOptions> = params.initialization_options
+                              .and_then(|options| serde_json::from_value(options)
```

Comment:
I think it is fair to assume that clients will pass valid JSON. They might add options or have options  missing and we should handle that gracefully, but if the send malformed JSON, then I think they are not sticking to the LSP and I hope that doesn't happen to often (and the extension developers should hit this before users do).